src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,106 +69,232 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
+   */
+  const findParentNode = () => {
+    if (!currentNodeId.current) {
+      return;
+    }
+    const parentId = visibleNodes.find((el) => el.id === currentNodeId.current).parent;
+    if (parentId !== '') {
+      cursor.current = visibleNodes.findIndex((el) => el.id === parentId);
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  /**
+   * Finds the first node starting with the given character.
+   * Starts at the cursors current position, and wraps around to the beginning of the menu if a match isn't found.
+   * @param {*} char The character to search by
+   */
+  const findNodeMatching = (char) => {
+    let sortedNodes = visibleNodes.slice(cursor.current + 1, visibleNodes.length);
+    sortedNodes = sortedNodes.concat(visibleNodes.slice(0, cursor.current));
+    const match = sortedNodes.find((el) => el.id[0].toUpperCase() === char);
+    if (match) {
+      cursor.current = visibleNodes.findIndex((el) => el.id === match.id);
+      currentNodeId.current = match.id;
+      focusCurrentNode();
+    }
+  };
+
   const handleKeyDown = (event, item) => {
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      event.preventDefault();
-      handleOnClick(event, item);
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        event.preventDefault();
+        handleOnClick(event, item);
+        break;
+      case KeyCode.KEY_DOWN:
+        event.preventDefault();
+        handleDownArrow();
+        break;
+      case KeyCode.KEY_UP:
+        event.preventDefault();
+        handleUpArrow();
+        break;
+      case KeyCode.KEY_RIGHT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'true') {
+            handleDownArrow();
+          } else if (document.getElementById(currentNodeId.current).getAttribute('aria-haspopup') && (!document.getElementById(currentNodeId.current).getAttribute('aria-expanded') || document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'false')) {
+            handleOnClick(event, item);
+          }
+        }
+        break;
+      case KeyCode.KEY_LEFT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'true') {
+            handleOnClick(event, item);
+          } else if (!document.getElementById(currentNodeId.current).getAttribute('aria-haspopup') || !document.getElementById(currentNodeId.current).getAttribute('aria-expanded') || document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'false') {
+            findParentNode();
+          }
+        }
+        break;
+      case KeyCode.KEY_HOME:
```

Comment:
It appears that the home and end keys no longer function. It's likely an artifact of the previous state refactor.

Reply:
Yep, good catch. That got factored out when I reduced the number of re renders used by the component. I've fixed it now.